File: RedSheet.DbEntities/Models/vLockRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vLockRecords", Schema = "dbo")]
	[Serializable]
	public partial class vLockRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ApplicationModuleId { get; set; } 
 
	[MaxLength(100)]
	public string ChildModuleName { get; set; } 
 
	[Required]
	public System.DateTime ExpiresAt { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int LockRecordId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RecordId { get; set; } 

	public vLockRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-lock-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vLockRecord {
    constructor(vLockRecord?: vLockRecord )  {
        let properties = [ "applicationModuleId", "childModuleName", "expiresAt", "lockRecordId", "recordId",];
        for (let property of properties)
            if (vLockRecord && vLockRecord[property])
                this[property] = vLockRecord[property];
    }
 
	applicationModuleId : number =   0 ;
 
    @maxLength(100)
	childModuleName : string =   undefined;
 
    @required()
	expiresAt : Date =   undefined;
 
    @range(1,2147483647)
	lockRecordId : number =   undefined;
 
    @range(1,2147483647)
	recordId : number =   undefined;


}
